lib/src/simple_op_store.rs

```diff
@@ -43,34 +46,150 @@ impl From<PersistError> for OpStoreError {
     }
 }
 
-impl From<protobuf::Error> for OpStoreError {
-    fn from(err: protobuf::Error) -> Self {
+impl From<thrift::Error> for OpStoreError {
+    fn from(err: thrift::Error) -> Self {
         OpStoreError::Other(err.to_string())
     }
 }
 
+// TODO: In version 0.7.0 or so, inline ThriftOpStore into this type and drop
+// support for upgrading from the proto format
 #[derive(Debug)]
 pub struct SimpleOpStore {
-    path: PathBuf,
+    delegate: Box<dyn OpStore>,
+}
+
+#[cfg(feature = "legacy_protobuf")]
+fn upgrade_to_thrift(store_path: PathBuf) -> std::io::Result<()> {
```

Comment:
Is this intended to be retriable on failure?

Reply:
I'll mark this resolved, but feel free to reopen if you think we should be more careful.

Reply:
Considering the early, unstable stage of things, some compromise seems fine if it's manually recoverable :+1: